pkg/daemon/ceph/osd/volume.go

```diff
@@ -237,6 +237,12 @@ func (a *OsdAgent) initializeDevices(context *clusterd.Context, devices *DeviceO
 		if device.Data == -1 {
 			logger.Infof("configuring new device %s", name)
 			deviceArg := path.Join("/dev", name)
+			// ceph-volume prefers to use /dev/mapper/<name> if the device has this kind of alias
+			for _, devlink := range device.PersistentDevicePaths {
+				if strings.HasPrefix(devlink, "/dev/mapper") {
```

Comment:
We require the `/dev/mapper` prefix wherever the devlink is applied? Otherwise it will be ignored? If so, why are the doc examples using other paths rather than `/dev/mapper`?

Reply:
Using the `/dev/mapper`-prefixed argument is necessary in certain cases,
regardless of whether or not to use `devLinksFilter: "/dev/mapper/..."`.

`ceph-volume` accepts only a device name under `/dev/mapper` if the device has
that type of alias.
(See [disk.py](https://github.com/ceph/ceph/blob/v14.2.4/src/ceph-volume/ceph_volume/util/disk.py#L772).)

On the other hand, the current version of Rook calls `ceph-volume lvm batch` with `/dev/<KNAME>`.
This causes an error with the device mapper.
For example, if `dm-0` has an alias like `/dev/mapper/crypt-sda`, Rook invokes `ceph-volume`
with an argument of `/dev/dm-0`, but `ceph-volume` returns an error.

This will happen even when using a device filter like `deviceFilter: "dm-.*"`.

So, in this PR, I have arranged the codes to pass `/dev/mapper` name to `ceph-volume` if necessary.